peps/pep-0817.rst

```diff
@@ -0,0 +1,2528 @@
+are usually installed with. Problems with installing PyTorch
+are a very common point of user confusion. To quantify this, on
+2025-12-05, 552 out of 8136, or 6.8%, of issues on `uv's issue tracker
+<https://github.com/astral-sh/uv/issues/>`__ contained the term "torch".
+
+Wheel variants remove this special casing and make GPU and TPU packages
+work just as well as regular packages with native code. They also reduce
+the burden on the maintainers to run separate infrastructure and to find
+and use non-standard features such as local version segments present on
+an index.
+
+**Induced Security Risk:** This approach has unfortunately led to supply
+chain attacks - more details on the `PyTorch Blog
+<https://pytorch.org/blog/compromised-nightly-dependency/>`__. It's a
+non-trivial problem to address which has forced the PyTorch team to
+create a complete mirror of all their dependencies, and is one of the
+core motivations behind :pep:`766`.
+
+The complexity of configuration often leads to projects providing ad-hoc
+installation instructions rather than covering permanent settings. This
+can lead to users being unable to cleanly upgrade the packages, or the
+upgraded packages being reverted to the default variant on subsequent
+upgrades.
+
+
+Package names as variants
+'''''''''''''''''''''''''
+
+Packages such as `XGBoost
+<https://xgboost.readthedocs.io/en/stable/install.html>`__ use different
+package names to approximate variants:
+
+.. code:: bash
+
+    pip install xgboost      # NVIDIA GPU variant
+    pip install xgboost-cpu  # CPU-only variant
+
+Maintainers of other software cannot express that they depend on either
+of the available variants being selected. They need to
+either depend on a specific variant, provide multiple alternative
+dependency sets using extras, or even publish their own software using
+multiple package names matching upstream variants.
+
+Commonly, these packages install overlapping files. Since Python
+packaging does not support expressing that two packages are mutually
+exclusive, installers can install both of them to the same environment,
+with the package installed second overwriting files from the one
+installed first. This leads to runtime errors, and
+the possibility of incidentally switching between variants depending on
+the way package upgrades are ordered.
+
+An additional limitation of this approach is that publishing a new
+release synchronously across multiple package names is not currently
+possible. :pep:`694` proposes adding such a mechanism for multiple
+wheels within a single package, but extending it to multiple packages is
+not a goal.
+
+**Induced Security Risk:** proliferation of suffixed variant packages
+leads users to expect these suffixes in other packages, making name
+squatting much easier. For example, one could create a malicious
+``numpy-cuda`` package that users will be lead to believe it's a CUDA
+variant of NumPy.
+
+`CuPy <https://cupy.dev/>`_ had to build
```

Comment:
I assume building 52 packages is per release rather than just once in the project history
```suggestion
As at the time of writing, `CuPy <https://cupy.dev/>`__ has to build
```

Reply:
No, it's a sum total of all packages ever built (until the time of writing). At any particular time, a different range of packages was built, so every package covers a different range of versions, and most of them are no longer updated.